src/decide.proto

```diff
@@ -0,0 +1,78 @@
+syntax = "proto3";
+import "google/protobuf/empty.proto";
+
+package decide;
+
+/* The payload for a requested state change to a component. Component state may
+   be a scalar (int32) or a label (string) */
+message StateChange {
+  string component = 1;
+  oneof state {
+    int32 scalar = 4;
+    string label = 5;
+  }
+}
+
+/* The payload for a requested change to the parameters for a component */
+message ComponentParams {
+  string component = 1;
+  // The documentation for the component must define how the parameter value will be parsed from the string.
+  map<string, string> parameters = 2;
+}
+
+/* This is what gets sent on the wire */
+message Request {
+  string protocol = 1;
+  oneof request {
+    // request change to state of a component. Reply type is ok or error.
+    StateChange state_change = 2;
+    // request reset to default state; value is name of component. Reply type is ok or error
+    string state_reset = 3;
+    // request lock on experiment; value is name of experiment. Reply type is ok or error
+    string lock_expt = 16;
+    // request unlock of experiment. Reply type is ok or error
+    google.protobuf.Empty unlock_expt = 17;
+    // request shutdown of controller. Reply type is ok or error
+    google.protobuf.Empty shutdown = 18;
+    // request update to parameter values. Reply type is ok or error.
+    ComponentParams set_parameters = 19;
+    // request parameter values for component. Reply type is params or error.
+    string get_parameters = 20;
+    // request config values for component. Reply type is config or error.
+    string get_component_config = 21;
+    // request config identifier for the controller. Reply type is config or error.
+    google.protobuf.Empty get_controller_config = 22;
+  }
+}
+
+/* These are the reply types */
+message Reply {
+  string protocol = 1;
+  oneof reply {
+    // For state_change, state_reset, lock_expt, unlock_expt, shutdown:
+    // indicates the request was correctly formed and was acted on
+    google.protobuf.Empty ok = 2;
+    // indicates an error with the request, contents give the cause
+    string error = 3;
```

Comment:
My thought was that errors are always fatal for the client, because they only get returned for invalid requests, or invalid assumptions about the configuration or state of the controller. But if you can think of some errors that the client ought to handle, we can expand this to include an enum.